app/routes/individual_response.py

```diff
@@ -0,0 +1,156 @@
+from flask import Blueprint, g, redirect, request, url_for
+from flask_login import current_user, login_required
+from structlog import get_logger
+
+from app.authentication.no_token_exception import NoTokenException
+from app.globals import get_metadata, get_session_store
+from app.helpers.language_helper import handle_language
+from app.helpers.schema_helpers import with_schema
+from app.helpers.session_helpers import with_questionnaire_store
+from app.helpers.template_helper import render_template
+from app.utilities.schema import load_schema_from_session_data
+from app.views.handlers.individual_response import (
+    IndividualResponseHandler,
+    IndividualResponseHowHandler,
+    IndividualResponsePostAddressConfirmHandler,
+)
+from app.questionnaire.location import InvalidLocationException
+
+logger = get_logger()
+
+individual_response_blueprint = Blueprint(
+    name="individual_response", import_name=__name__, url_prefix="/individual-response/"
+)
+
+
+@individual_response_blueprint.before_request
+def before_individual_response_request():
+    metadata = get_metadata(current_user)
+    if not metadata:
+        raise NoTokenException(401)
+
+    logger.bind(
+        tx_id=metadata["tx_id"],
+        schema_name=metadata["schema_name"],
+        ce_id=metadata["collection_exercise_sid"],
+        questionnaire_id=metadata["questionnaire_id"],
+    )
+
+    logger.info(
+        "individual-response request", method=request.method, url_path=request.full_path
+    )
+
+    handle_language()
+
+    session_store = get_session_store()
+    g.schema = load_schema_from_session_data(session_store.session_data)
+
+
+@individual_response_blueprint.route("/", methods=["GET", "POST"])
+@login_required
+@with_questionnaire_store
+@with_schema
+def request_individual_response(schema, questionnaire_store):
```

Comment:
This route and all IR routes need to check if the list store for the IR list is empty. At the moment, you can access these pages even when the list is empty, which results in 500 errors.